### ISSUE TITLE
Standalone optional 'install'

### ISSUE THREAD

**Issue by vanhoutenbos** [louislam/uptime-kuma#3332]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

We would like to be able to not check for updates / install on startup of the .exe
This is motivated by 2 things;
1. We would like to optionally update, sometimes we restart the machine and we get a free update of kuma which is not always desired
2. on one of our machines we do not have internet access (for example going to github.com and cloning the repo)

For the machine that doesn't have internet we 'install' is on a local machine, and than zip the package and just 'run' the executable.

### ✔️ Solution

Could this be an optional param for example `uptime-kuma.exe installOrUpdate=false`

### ❓ Alternatives

An alternative could the to add a way to send the internet calls through a proxy

**Comment by CommanderStorm** [comment]
Code is located under https://github.com/louislam/uptime-kuma/tree/eaee55fc8f18de4511cbc17486daf839f4ac867b/extra/exe-builder
PRs are welcome for additional functionality of the exe

**Comment by CommanderStorm** [comment]
I don't fully understand your need, given that
-  your machine has to contact the internet for notifications anyway => just disabling updates will not be a solution
- if you want immutable infrastructure, I would argue that the docker image is a better fit => just disabling updates will not be a solution

**Comment by CommanderStorm** [comment]
> I don't fully understand your need, given that
> 
> * your machine has to contact the internet for notifications anyway
> * if you want immutable infrastructure, I would argue that the docker image is a better fit
> 
> For both cases just disabling updates will likely not be a solution

I think this is important. If what you are proposing is a solution is something we have not discussed until now.

**Comment by vanhoutenbos** [comment]
Is it open for discussion? I would be open to help make the feature since my teams desire this quite a bit for it to work on our VMs 😁

**Comment by CommanderStorm** [comment]
Yes, this is open for discussion.
That Is why I said that I think we should discuss if an option like his would solve your problem. 
Sorry if there was some misunderstanding.

I am currently not quite convinced, that disabling updates in the executable will solve/is the best solution for your issue.

**Comment by vanhoutenbos** [comment]
Ah, my apologies I misread it!

What the goal is is that we can use uptime kuma without access to the Internet.

We have a intranet that we want to monitor using the tool but do not have the option to for example use `npm i`

**Comment by CommanderStorm** [comment]
This does not answer the questions.

Let me rephrase:
- How do you plan to get notifications out, if the system is airgapped?
- Why is the executable the right tool? (in comparison to the docker image). If you want immutable airgapped infrastructure, I would argue that is the better tool.

**Comment by vanhoutenbos** [comment]
Good questions, some background first: the system we are using is very old, simple windows server 2012 (I believe) servers, not airgapped specifically but all ports are closed for Internet traffic (non-vpn traffic).

Though some services have been opened to support the current software in place like email, so that we can email when something is wrong.
To get more to be opened its a slow and cumbersum process.

We are also migrating away to a more modern (openshift) infrastructure.
The old machines communicate with the modern apps but need to be monitored, that's where kuma comes in.


As for the questions:
How do you plan to get notifications out, if the system is airgapped?
We can send emails from the server over smtp and have some urls open to send something over we hooks. 

Why is the executable the right tool? (in comparison to the docker image). If you want immutable airgapped infrastructure, I would argue that is the better tool.
Unfortunately installing docker on the machine would be a long and cumbersome process and also a bit overkill for just this tool. 
In the end we would still have issues getting the image on the machine etc. 

Tl:dr the lesser time consuming solution would be to either run the standalone and remove any network access needed or find a different method without needing to install docker or something

**Comment by CommanderStorm** [comment]
Agree, installing docker on something which is so ancient (11 years ago is a long time) is probably not as simple as just following the installer.

The other alternative I had in mind (Copying over the node packages necessary) is as well less likely to work as nodejs would need to be installed and installing something on an semi-airgapped system is a pain.

**Comment by vanhoutenbos** [comment]
This is true indeed, but maybe figuring this out is easier than changing the workings of the executable.

I am open to either solution.

I might giving the nodejs version a try, I have been thinking about it since installing node on an ancient machine might be easier than figuring out how to get around the automated update.

I understand from the products perspective that adding such code for a niche use case might not be worth it that's also why I am asking if making a PR for this might be a waste of my time or not 😁

**Comment by vanhoutenbos** [comment]
I made a PR, we are using the code from this PR in our machine, the first of 4 environments (Test) works flawless :)
